### ISSUE TITLE
Photos in frames have excessive shimmering artifacts (no mipmaps)

### ISSUE THREAD

**Issue by bravely-beep** [mortuusars/Exposure#131]
Photographs displayed in frames do not use mipmaps, causing severe artifacts when viewed from a distance. This is true for vanilla minecraft maps as well; however, the dithering used in Exposure's photographs exacerbates the issue considerably, causing photographs to shimmer excessively even when viewed from relatively close distances.

[This mod](https://github.com/Jalvaviel/MapMipMapMod) seems to resolve the issue for vanilla maps; perhaps a similar solution could be used here?

(Using `exposure-1.20.1-1.7.7-fabric.jar`)

**Comment by mortuusars** [comment]
I was trying to fix this one day, but to no avail.
Thanks for pointing out this mod, it's exactly what I was trying to find.
My only hope is that it'll work, because I have read somewhere that mipmapping cannot be enabled for non power of 2 texture sizes, and exposure uses non-standard texture size of 320x320.

**Comment by mortuusars** [comment]
Yep, it works! This is huge, it looks so much better now, thank you so much for sending me mapmipmapmod repo. And thanks to Jalvaviel for implementation example.
This will be part of a 1.21 port, but I'll try to backport it to prior versions as well.

**Comment by mortuusars** [comment]
Added in 1.21.1 - 1.8 and 1.20.1-1.7.8.
